lib/Models/RegionMapping.js

```diff
@@ -384,6 +384,12 @@ RegionMapping.prototype.loadRegionDetails = function() {
     });
 };
 
+// This is a test interface shim to provide test access to calculateImageryLayerIntervals(timeColumn, currentTime, animateForwards, cyclic).
+// This should only ever be called from test cases.
+RegionMapping.prototype._testCalculateImageryLayerIntervals = function(timeColumn, currentTime, animateForwards, cyclic) {
```

Comment:
I think the cleanest way to do this is to break the `calculateImageryLayerIntervals` function into a separate module, and then require it in here and in the tests.  You can mark it `@private` in the jsdoc comments so it doesn't show up in the documentation.